### DISCUSSION TITLE
Can merging PDFs be made faster and simpler?

### DISCUSSION THREAD

**Discussion by ysageev** [Hopding/pdf-lib#1039] [Ideas]
Hi!  First, let me tell you how grateful I am for pdf-lib.  I'm finding it full featured and reliable.  Thank you for making this available as FOSS.

My one issue is that merging PDFs is slow, at least the way I'm doing it.  When I send out reports, I have to append a bunch of related supporting PDFs, using this code:

```
const mergePDF = async (pdfbuf, pdfdest) => {
   let pdfDoc = await PDFDocument.load(pdfbuf, { ignoreEncryption: true, returnCopy: true })
   let copiedPages = await pdfdest.copyPages(pdfDoc, pdfDoc.getPageIndices())
   copiedPages.forEach((page) => {
      pdfdest.addPage(page)
   })
   return pdfdest
}
```

Right now, we need to call `addPage()` for every page in `copiedPages`.  Why couldn't the API have something `addPages(copiedPages)`?  

Better yet, because merging an entire PDF is so common, why can't there be an API addition like `pdfdst.merge(pdfsrc)` which is highly optimized?

**Comment by StillOnMyWay** [comment]
This is exactly what I came here to ask Andrew - @Hopding . I figured 💯 % that this was an intentional decision not to add this feature, because you've already created `embedPages `and `copyPages`, so I was wondering why no `addPages`

**Comment by acestronautical** [comment]
The slow speed of PDF merging is a bottleneck for us over at https://github.com/momijizukamori/bookbinder-js

It's especially frustrating that this can't be done with promises.
